cbox/cliparser.py

```
@@ -30,11 +32,20 @@ def get_cli_parser(func, skip_first=0, parser=None):
             continue
 
         if arg_default is not _empty:
-            parser.add_argument(
-                arg_name, type=arg_type, default=arg_default,
-                required=arg_required, help=arg_help
-            )
+            if arg[1] == bool:
+                action = 'store_{}'.format(str(not arg_default).lower())
+                parser.add_argument(
+                    arg_name, default=arg_default,
+                    required=arg_required, help=arg_help, action=action
+                )
+            else:
+                parser.add_argument(
+                    arg_name, type=arg_type, default=arg_default,
+                    required=arg_required, help=arg_help
+                )
         else:
+            if arg[1] == bool:
```

this is a breaking change, we should handle the `bool type required argument` case.
Please support things like `python mycli.py --my-bool <value>` where the `<value>` may be any reasonable option.

I think yaml's bool supported values are good idea - https://yaml.org/type/bool.html. the regex is:

```python
bool_value_regex = re.compile(r'^(y|Y|yes|Yes|YES|n|N|no|No|NO|true|True|TRUE|false|False|FALSE|on|On|ON|off|Off|OFF)$')
```

or 

```python
bool_value_regex = re.compile(r'^(y|yes|n|no|true|false|on|off)$', flags=re.I)
```

you can also add '1' and '0'

in case of invalid value we should raise an exception.